cpp/src/arrow/util/hashing.h

```diff
@@ -690,7 +690,8 @@ class BinaryMemoTable : public MemoTable {
     DCHECK_LE(start, size());
 
     const builder_offset_type* offsets = binary_builder_.offsets_data();
-    const builder_offset_type delta = offsets[start];
+    const builder_offset_type delta =
+        start < binary_builder_.value_data_length() ? offsets[start] : 0;
```

Comment:
I made a mistake here, `binary_builder_.value_data_length()` should be `binary_builder_.length()`

Reply:
Can you post a followup PR (just reuse the same JIRA number)?